### ISSUE TITLE
XmlMapper seems to ignore custom serializer

### ISSUE THREAD

**Issue by matejj** [FasterXML/jackson-dataformat-xml#41]
I tried to add custom serializer to my XmlMapper for XML serialization: 

```
JacksonXmlModule module = new JacksonXmlModule();
module.addSerializer(Item.class, new ItemSerializer());
XmlMapper xml = new XmlMapper(module);
```

However, it seems like XmlMapper ignore my serializer and serialize code on its own. 

I tried this same serializer with ObjectMapper for JSON serialization:

```
SimpleModule module = new SimpleModule("MyModule", new Version(1,0,0,null,null,null);
module.addSerializer(Item.class, new ItemSerializer());
ObjectMapper json = new ObjectMapper();
json.registerModule(module);
```

It works fine for JSON.

**Comment by felixWackernagel** [comment]
Still open. I write
```
final ObjectMapper mapper = new XmlMapper();
final SimpleModule = new JacksonXmlModule();
module.addSerializer(Item.class, new ItemSerializer());
mapper.registerModule( module );
```
And code a Infinite recursion (StackOverflowError).
I test it in 2.4.5 and 2.5.1

**Comment by cowtowncoder** [comment]
@felixWackernagel There is unit test (`TestSerializerCustom.java`) that checks:

```java
    public void testIssue42() throws Exception
    {
        XmlMapper xmlMapper = new XmlMapper();
        SimpleModule m = new SimpleModule("module", new Version(1,0,0,null,null,null));
        m.addSerializer(Item.class, new ItemSerializer());
        m.addDeserializer(Item.class, new ItemDeserializer());
        xmlMapper.registerModule(m);
        ....
```

so I would need more information on how your case differs.

But for issue tracking purposes, could you open a new issue? It is likely that underlying cause is different, and it is easier to create release notes when fixes are not spread across multiple releases (that is, fix for this problem was in 2.0.7; fix for remaining issue would go in 2.5.2)

**Comment by kbansod** [comment]
We have upgraded to jersey 2.29.1 and so jackson to lastest 2.10.0. It seems like XmlMapper still ignores custom serializer.

My code is exactly as below. It is being invoked via XMLContextResolver. But _serializers = null in XmlMapper object. 

JacksonXmlModule module = new JacksonXmlModule();
module.addSerializer(Item.class, new ItemSerializer());
XmlMapper xml = new XmlMapper(module);

It seems XmlMapper still ignores custom serializer.

I tried using ObjectMapper with SimpleModule, but it does give me desired xml. I get desried xml using JacksonXmlModule, but I want to do some additional work using custom serializer. 

Can someone help me enforce custom serializer in above code? I want to use JacksonXmlModule only.
